numpy/core/tests/test_multiarray.py

```diff
@@ -1731,6 +1731,19 @@ def test_transpose(self):
         assert_raises(ValueError, lambda: a.transpose(0, 0))
         assert_raises(ValueError, lambda: a.transpose(0, 1, 2))
 
+    def test_sort_by(self):
+        msg = "Test sorting by absolute values as keys"
+        a = np.array([-2, -3, -4, -1, -6])
+        b = np.sort(a, by=(np.absolute(a),))
+        e = np.array([-1, -2, -3, -4, -6])
+        assert_equal(b, e, msg)
+        # Test sorting complex numbers by absolute value
+        msg = "Test sorting comple xnumbers by absolute value"
+        carr = np.arange(4, dtype=np.complex128)[::-1].reshape(2, 2)
+        carr_out = np.array([[2+0j, 3+0j], [0+0j, 1+0j]], dtype=np.complex128)
+        carr.sort(axis=1, by=np.absolute(carr))
```

Comment:
maybe also add the carr.real, carr.imag test mentioned above. Also, would help to check for different shapes of keys and mismatch between keys and arr shape for error assertions.